File: src/chapter1/ZuiChangLianXuBuChongFuZiXuLie/Main.java
package chapter1.ZuiChangLianXuBuChongFuZiXuLie;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * @author mbfjllybl
 * @version 1.0
 * @description: TODO
 * @date 2021/11/19 下午8:18
 */
public class Main {
    public static int[] count = new int[100010];
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine());
        String[] strs = reader.readLine().split(" ");
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(strs[i]);
        }
        int res = 0;
        for (int i = 0, j = 0; i < n; i++) {
            while (count[arr[i]] > 0) count[arr[j++]]--;
            count[arr[i]]++;
            res = Math.max(res, i - j + 1);
        }
        System.out.println(res);;
    }
}

File: src/chapter1/ZuiChangLianXuBuChongFuZiXuLie/Main.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 1000010;
int nums[N];

int main() {
    int n;
    cin >> n;
    int res = 0;
    unordered_map<int, int> u_map;
    for (int i = 0; i < n; i++) cin >> nums[i];
    for (int i = 0, j = 0; i < n; i++) {
        while (u_map.count(nums[i]) != 0 && u_map[nums[i]] != 0) {
            u_map[nums[j]]--;
            j++;
        }
        u_map[nums[i]]++;
        res = max(res, i - j + 1);
    }
    cout << res;
    return 0;
}